app/models/email_configuration.rb

```diff
@@ -0,0 +1,45 @@
+# frozen_string_literal: true
+
+class EmailConfiguration < ApplicationRecord
+
+  validates :email, :password, :user_name, :authentication, :domain, :tls, :address, length: { maximum: 255 }
+
+  validates :email, :password, :user_name, presence: true
+  validates :email, format: { with: Authentication::RE_EMAIL_OK, message: Authentication::MSG_EMAIL_BAD }
```

Comment:
I have a gem to do email validation, which was quite famous :) https://github.com/hallelujah/valid_email (contributors needed :) )
I am not sure about using this validation and how it plays with on-premises, I can imagine that people want to send them to internal aliases which do not have domain parts. Well it is not defined, it is for SaaS only so we can be at ease with validating like this first.

Reply:
I'm using what we already have in other places we set email addresses. I believe these need to stay consistent.

Your gem is awesome and I'm starring it to use later.